tests/device_handler.py

```diff
@@ -49,12 +54,36 @@ def __init__(self, client: "Client", nowait: bool = False) -> None:
     def _configure_client(self, client: "Client") -> None:
         self.client = client
         self.client.ui = NullUI  # type: ignore [NullUI is OK UI]
+        self.client.button_callback = self.client.ui.button_request
         self.client.watch_layout(True)
         self.client.debug.input_wait_type = DebugWaitType.CURRENT_LAYOUT
 
-    def run(
+    def run_with_session(
+        self,
+        function: t.Callable[tx.Concatenate["Session", P], t.Any],
+        seedless: bool = False,
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> None:
+        """Runs some function that interacts with a device.
+
+        Makes sure the UI is updated before returning.
+        """
+        if self.task is not None:
+            raise RuntimeError("Wait for previous task first")
+
+        # wait for the first UI change triggered by the task running in the background
+        if seedless:
+            session = self.client.get_seedless_session()
+        else:
+            session = self.client.get_session()
```

Comment:
IIUC, click tests may get stuck here, in case they are using DebugLink to enter PIN - so since `client.get_session()` doesn't know the PIN it won't be able to finish.